lib/OpenQA/Worker/Engines/isotovideo.pm

```diff
@@ -326,11 +338,22 @@ sub engine_workit {
         my $error = locate_local_assets(\%vars, $assetkeys);
         return $error if $error;
     }
+    my $casedir     = OpenQA::Utils::testcasedir($vars{DISTRI}, $vars{VERSION}, $shared_cache);
+    my $productdir  = OpenQA::Utils::productdir($vars{DISTRI}, $vars{VERSION}, $shared_cache);
+    my $target_name = path($casedir)->basename;
 
     $vars{ASSETDIR}   //= OpenQA::Utils::assetdir();
-    $vars{CASEDIR}    //= OpenQA::Utils::testcasedir($vars{DISTRI}, $vars{VERSION}, $shared_cache);
-    $vars{PRODUCTDIR} //= OpenQA::Utils::productdir($vars{DISTRI}, $vars{VERSION}, $shared_cache);
+    $vars{CASEDIR}    //= $target_name;
+    $vars{PRODUCTDIR} //= substr($productdir, rindex($casedir, $target_name));
+
+    if (my $error = _link_repo($casedir, $pooldir, $target_name)) { return $error }
 
+    if (   $vars{NEEDLES_DIR}
+        && !File::Spec->file_name_is_absolute($vars{NEEDLES_DIR})
+        && $vars{NEEDLES_DIR} !~ /http(s)?\:\/\//)
```

Comment:
Please add a comment explaining the speciality about "http" here.

Reply:
This is meant to check if `NEEDLES_DIR` is a relative path to a local folder, isn't it? (it's not absolute, it's not http or https). In that case it might be cleaner to check the opposite. See if this is a valid, relative folder that exists. Otherwise leave it alone.

Reply:
added the comments

Reply:
If I understand correctly, usually, we don't specify NEEDLE_DIRS here, only users need to get needles from URL address (github or gitlab), or users use `openqa-clone-custom-git-refspec`. For now, this script will give an absolute path as a NEEDLE_DIR, such as `NEEDLES_DIR=/var/lib/openqa/cache/openqa.suse.de/sle/products/sle/needles`. We don't need to do symlink in these scenarios. 
In the future, we will change the `NEEDLE_DIR=needles` in `openqa-clone-custom-git-refspec`, we need to do symlink at that time.

Reply:
That sounds good 👍🏽  My idea was just to look at the filesystem and see if the relative path exists - but maybe we get a better error message the way you implemented it. Thank you for expanding on your motivation.